

<!DOCTYPE html>
<!--[if IE 8]><html class="no-js lt-ie9" lang="en" > <![endif]-->
<!--[if gt IE 8]><!--> <html class="no-js" lang="en" > <!--<![endif]-->
<head>
  <meta charset="utf-8">
  
  <meta name="viewport" content="width=device-width, initial-scale=1.0">
  
  <title>10. Imputation &mdash; Shalek Lab HowTos  documentation</title>
  

  
  
  
  

  
  <script type="text/javascript" src="_static/js/modernizr.min.js"></script>
  
    
      <script type="text/javascript" id="documentation_options" data-url_root="./" src="_static/documentation_options.js"></script>
        <script type="text/javascript" src="_static/jquery.js"></script>
        <script type="text/javascript" src="_static/underscore.js"></script>
        <script type="text/javascript" src="_static/doctools.js"></script>
        <script type="text/javascript" src="_static/language_data.js"></script>
    
    <script type="text/javascript" src="_static/js/theme.js"></script>

    

  
  <link rel="stylesheet" href="_static/css/theme.css" type="text/css" />
  <link rel="stylesheet" href="_static/pygments.css" type="text/css" />
  <link rel="stylesheet" href="_static/css/custom.css" type="text/css" />
    <link rel="index" title="Index" href="genindex.html" />
    <link rel="search" title="Search" href="search.html" />
    <link rel="next" title="11. Clustering" href="clustering.html" />
    <link rel="prev" title="9. Batch Correction" href="batch_correction.html" /> 
</head>

<body class="wy-body-for-nav">

   
  <div class="wy-grid-for-nav">
    
    <nav data-toggle="wy-nav-shift" class="wy-nav-side">
      <div class="wy-side-scroll">
        <div class="wy-side-nav-search"  style="background: #F9F9F9" >
          

          
            <a href="index.html">
          

          
            
            <img src="_static/logo.png" class="logo" alt="Logo"/>
          
          </a>

          
            
            
          

          
<div role="search">
  <form id="rtd-search-form" class="wy-form" action="search.html" method="get">
    <input type="text" name="q" placeholder="Search docs" />
    <input type="hidden" name="check_keywords" value="yes" />
    <input type="hidden" name="area" value="default" />
  </form>
</div>

          
        </div>

        <div class="wy-menu wy-menu-vertical" data-spy="affix" role="navigation" aria-label="main navigation">
          
            
            
              
            
            
              <p class="caption"><span class="caption-text">Introduction</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="home.html">Welcome to the HowTos wiki!</a></li>
</ul>
<p class="caption"><span class="caption-text">Terra</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="how_to_for_new_terra_users.html">How to for new Terra Users</a></li>
<li class="toctree-l1"><a class="reference internal" href="how_do_i_use_terra_pipelines.html">How do I use Terra Pipelines?</a></li>
<li class="toctree-l1"><a class="reference internal" href="how_do_i_use_terra_notebooks.html">How do I use Terra Notebooks?</a></li>
<li class="toctree-l1"><a class="reference internal" href="drop_seq_tools_on_terra.html">Drop Seq Tools on Terra</a></li>
<li class="toctree-l1"><a class="reference internal" href="drop_seq_tools_on_terra.html#output-files">Output files:</a></li>
<li class="toctree-l1"><a class="reference internal" href="drop_seq_tools_on_terra.html#troubleshooting">Troubleshooting</a></li>
<li class="toctree-l1"><a class="reference internal" href="seqwell_demultiplexing_on_terra.html">SeqWell Demultiplexing on Terra</a></li>
<li class="toctree-l1"><a class="reference internal" href="kallisto_bustools_on_terra.html">Workflow</a></li>
<li class="toctree-l1"><a class="reference internal" href="kallisto_bustools_on_terra.html#notebook">Notebook</a></li>
</ul>
<p class="caption"><span class="caption-text">Data Analysis</span></p>
<ul class="current">
<li class="toctree-l1"><a class="reference internal" href="reading_in_the_data_to_r.html">1. Reading in the data (to R)</a></li>
<li class="toctree-l1"><a class="reference internal" href="dropout.html">2. Dropout?</a></li>
<li class="toctree-l1"><a class="reference internal" href="filtering.html">3. Filtering</a></li>
<li class="toctree-l1"><a class="reference internal" href="doublets.html">4. Doublets</a></li>
<li class="toctree-l1"><a class="reference internal" href="normalization.html">5. Normalization</a></li>
<li class="toctree-l1"><a class="reference internal" href="choosing_variable_genes.html">6. Choosing variable genes</a></li>
<li class="toctree-l1"><a class="reference internal" href="pca.html">7. PCA</a></li>
<li class="toctree-l1"><a class="reference internal" href="other_dimensionality_reduction.html">8. Other dimensionality reduction</a></li>
<li class="toctree-l1"><a class="reference internal" href="batch_correction.html">9. Batch Correction</a></li>
<li class="toctree-l1 current"><a class="current reference internal" href="#">10. Imputation</a></li>
<li class="toctree-l1"><a class="reference internal" href="clustering.html">11. Clustering</a></li>
<li class="toctree-l1"><a class="reference internal" href="clustering.html#how-do-i-know-i-have-a-good-clustering">12. How do I know I have a good clustering?</a></li>
<li class="toctree-l1"><a class="reference internal" href="clustering.html#what-if-i-am-looking-for-rare-cell-subsets">13. What if I am looking for rare cell subsets?</a></li>
<li class="toctree-l1"><a class="reference internal" href="clustering.html#what-if-some-cells-are-clearly-placed-in-the-wrong-clusters">14. What if some cells are clearly placed in the wrong clusters?</a></li>
<li class="toctree-l1"><a class="reference internal" href="differential_expression.html">15. Differential Expression</a></li>
<li class="toctree-l1"><a class="reference internal" href="cell_ordering_pseudotime.html">16. Cell Ordering (“Pseudotime”)</a></li>
<li class="toctree-l1"><a class="reference internal" href="network_inference.html">17. Network Inference</a></li>
<li class="toctree-l1"><a class="reference internal" href="i_have_a_bunch_of_genes__now_what.html">18. I have a bunch of genes  now what?</a></li>
<li class="toctree-l1"><a class="reference internal" href="meta_analysis.html">19. Meta analysis</a></li>
</ul>

            
          
        </div>
      </div>
    </nav>

    <section data-toggle="wy-nav-shift" class="wy-nav-content-wrap">

      
      <nav class="wy-nav-top" aria-label="top navigation">
        
          <i data-toggle="wy-nav-top" class="fa fa-bars"></i>
          <a href="index.html">Shalek Lab HowTos</a>
        
      </nav>


      <div class="wy-nav-content">
        
        <div class="rst-content">
        
          















<div role="navigation" aria-label="breadcrumbs navigation">

  <ul class="wy-breadcrumbs">
    
      <li><a href="index.html">Docs</a> &raquo;</li>
        
      <li>10. Imputation</li>
    
    
      <li class="wy-breadcrumbs-aside">
        
            
            <a href="_sources/imputation.md.txt" rel="nofollow"> View page source</a>
          
        
      </li>
    
  </ul>

  
  <hr/>
</div>
          <div role="main" class="document" itemscope="itemscope" itemtype="http://schema.org/Article">
           <div itemprop="articleBody">
            
  <div class="section" id="imputation">
<h1>10. Imputation<a class="headerlink" href="#imputation" title="Permalink to this headline">¶</a></h1>
<p>People discuss, as a way to compensate for dropout, imputing expected expression values of genes. The idea behind imputation is to use the fact that we assume that genes do not act in isolation - modules of genes move together - to share information across genes and cells.</p>
<p>Some people (read: Alex Shalek) object to this because it may remove real biological variation.</p>
<p>People in the wider computational sc world also have some issues with imputation, specifically a lack of benchmarks/comparisons https://github.com/theislab/scanpy/issues/189</p>
<p>Some imputation algorithms: <a class="reference external" href="https://github.com/KrishnaswamyLab/MAGIC">MAGIC</a>, CIDR, <a class="reference external" href="https://github.com/Vivianstats/scImpute">scImpute</a>, <a class="reference external" href="https://github.com/mohuangx/SAVER">SAVER</a></p>
<p>Sarah did a comp subgroup on Magic and SAVER - slides are <a class="reference external" href="https://docs.google.com/presentation/d/13Lb8-BgCp-5yxz9-O1DLcjVKAef3VcWCKtsID4-3RTM/edit?usp=sharing">here</a> as well as in the dropbox at: Talks &amp; Posters/Journal Club/comp_subgroup_JC_7_17_18.pptx</p>
<p>Sarah’s favorite between Magic and SAVER is SAVER because it takes into account the uncertainty of the imputation when estimating gene-gene correlation. Magic does describe a method to compute gene-gene associations in the paper, but the code is not available. Using pearson or spearman correlation to compute gene-gene correlations on the output of Magic imputations gives correlation between all genes of basically 0 or 1 indicating that this method might be over-imputing.</p>
<p>In general, Sarah thinks that imputation is a useful exercise in thinking about ways to model single cell data but should not necessarily be used to transform data then do downstream analysis. This is because, the zeros do contain some information and replacing them with imputed values without placing some uncertainty on these values loses this information. The insights from the development of imputation methods could allow for the use of these models in conjunction with later analysis as long as the uncertainty is propagated. If you use imputed data for downstream analysis, always check that you can still see whatever biological conclusion you come to in your non-imputed data. Imputation can be used to dig through the noise but should not be assumed to be correct!</p>
<p>This review https://ieeexplore.ieee.org/document/8388285 compares imputation methods.</p>
</div>


           </div>
           
          </div>
          <footer>
  
    <div class="rst-footer-buttons" role="navigation" aria-label="footer navigation">
      
        <a href="clustering.html" class="btn btn-neutral float-right" title="11. Clustering" accesskey="n" rel="next">Next <span class="fa fa-arrow-circle-right"></span></a>
      
      
        <a href="batch_correction.html" class="btn btn-neutral float-left" title="9. Batch Correction" accesskey="p" rel="prev"><span class="fa fa-arrow-circle-left"></span> Previous</a>
      
    </div>
  

  <hr/>

  <div role="contentinfo">
    <p>
        &copy; Copyright 2020, Shalek Lab

    </p>
  </div>
  Built with <a href="http://sphinx-doc.org/">Sphinx</a> using a <a href="https://github.com/rtfd/sphinx_rtd_theme">theme</a> provided by <a href="https://readthedocs.org">Read the Docs</a>. 

</footer>

        </div>
      </div>

    </section>

  </div>
  


  <script type="text/javascript">
      jQuery(function () {
          SphinxRtdTheme.Navigation.enable(true);
      });
  </script>

  
  
    
   

</body>
</html>